### DISCUSSION TITLE
Want my all commits in graph

### DISCUSSION THREAD

**Discussion by ghias70** [community/community#28781] [Other Feature Feedback, Questions, & Ideas]
in recent few months my user name and email was changed unintentionally from git config ,due to which all those commits are not showing in my contribution graph, can I bring all those previous commits back to my graph ?

**Comment by airtower-luna** [comment]
If I understand correctly, you changed the email address in your Git config? That means the new address gets recorded in all your commits as the author email. To get those commits assigned to your account you need to add the new email address to your account.

**Reply by ghias70** [reply]
> If I understand correctly, you changed the email address in your Git config? That means the new address gets recorded in all your commits as the author email. To get those commits assigned to your account you need to add the new email address to your account.

that email was not a valid email it was like "ffgf "

**Reply by airtower-luna** [reply]
Then your only option is to fix the commits so they contain a valid email address of yours (or your GitHub-noreply address, if you want to keep your email private). The problem is that it requires rewriting history, which will cause issues if you have already shared the commits with anyone else: All the commit IDs since the first affected commit will change, merges will have weird results, and fetches/pulls suddenly won't match. If the commits are already merged into someone else's repository assume it's not going to be possible to change them.

For your own repositories, check [Git Tools – Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). At the very end of the chapter there's even an [example of how to replace author data in all commits with a specific author email](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_changing_email_addresses_globally).
